### PR TITLE
refactor(rxNotify): FRMW-401 convert to quark service.

### PR DESCRIPTION
**Pull request by 100stacks**
JIRA: https://jira.rax.io/browse/FRMW-401

- [x] Dev LGTM
- [x] Dev LGTM
- [x] Design LGTM

---
Refactor `rxNotify` service to Quark service.

### CODE REVIEW

**Comment by 100stacks** [comment]
@CITguy, refactoring `rxNotify` _service_ into a Quark service appears to have a conflict with the `rxNotify` _component_.  The `rxNotify` _component_ **demo page** and **API Docs** is pulled in for the `rxNotify` _Quark service_.  One possibility is that they have identical names.  Any recommendations?


---
### `rxNotify` Component rendered on `rxNotify` Quark Service page
screen shot 2015-12-28 at 10 41 10 am

**Comment by CITguy** [comment]
@100stacks, yeah, it's because they have the same name. In `demoApp.js`, the routes resolve the module based on the name alone, so `/quarks/rxNotify` and `/atoms/rxNotify` both match on the name of the module, but the `_.find()` resolves to the first module found (the Component). 

To correct this, you'll need to tweak the routing logic.

* https://github.com/rackerlabs/encore-ui/blob/master/demo/scripts/demoApp.js#L140-L146
  * needs `category: "quarks"` added to `_.find()` arguments
* https://github.com/rackerlabs/encore-ui/blob/master/demo/scripts/demoApp.js#L170-L176
  * needs `category: "atoms"` added to `_.find()` arguments
* https://github.com/rackerlabs/encore-ui/blob/master/demo/scripts/demoApp.js#L200-L206
  * needs `category: "molecules"` added to `_.find()` arguments
* https://github.com/rackerlabs/encore-ui/blob/master/demo/scripts/demoApp.js#L218-L224
  * needs `category: "components"` added to `_.find()` arguments

**Comment by CITguy** [comment]
... additionally, `grunt-tasks/phantomjs-check.js` would need to be updated to use the full path instead of the generic `/modules/:module` path.

**Comment by 100stacks** [comment]
@CITguy, sounds good.  I noticed other Quarks with the same name as the Component.  I'll update the build scripts as provided above before proceeding with refactoring them.

File: src/quarks/quarks.module.js [hunk 1]
```diff
@@ -26,5 +26,6 @@
  * * {@link quarks.service:rxNestedElement rxNestedElement}
  * * {@link quarks.service:Environment Environment}
  * * {@link quarks.service:rxPromiseNotifications rxPromiseNotifications}
+ * * {@link quarks.service:rxNotify rxNotify}
```
Comment (CITguy) [inline src/quarks/quarks.module.js]: Please remove the change to this file. We'll revisit updating this file once all Quarks have been migrated.

**Comment by 100stacks** [comment]
@CITguy okay reverted updates to `quarks.module.js` and rebased. :+1:

**Comment by CITguy** [comment]
LGTM

**Comment by glynnis** [comment]
Not giving a Design LGTM here just yet because I'm not sure rxNotify is a quark under the new IA, but still need to confirm.

**Comment by CITguy** [comment]
@glynnis: this PR is for the `rxNotify` service, not the directive.

File: src/components/rxFeedback/rxFeedback.module.js [hunk 2]
```diff
@@ -126,9 +126,6 @@
  *
  * ## Directives
  * * {@link rxFeedback.directive:rxFeedback rxFeedback}
- *
```
Comment (100stacks) [inline src/components/rxFeedback/rxFeedback.module.js]: Maintenance: Remove artifact from Quarks refactoring.

**TODO:** Add  `{@link quarks.service:rxScreenshotSvc rxScreenshotSvc}` to `quarks.module.js`.

**Comment by glynnis** [comment]
@CITguy Gotcha. And I see we're using `rx-example` here, so concerns about duplication of visual demos are moot.

Design LGTM :+1:

**Comment by CITguy** [comment]
Actually, after taking another look at this PR, I wonder if we should pull the "Create Promise", "Resolve/Reject Promise" demo out of the _component_ and into its own self-contained demo. @100stacks, can you please update this PR to do this?

**Comment by CITguy** [comment]
@glynnis, you are correct. With this PR, we do have duplication of demo code. We have an example, but it's not being used yet.

**Comment by 100stacks** [comment]
@CITguy, For https://github.com/rackerlabs/encore-ui/pull/1474#issuecomment-170971546, we pulled the "Create Promise" demo out in the `rxPromiseNotifications` PR https://github.com/rackerlabs/encore-ui/pull/1476 ...keeping the original `rxNotify` component demo intact until we remove the component.  

Should I reference that example here, or breakout another section of the `rxNotify` component demo?

**Comment by CITguy** [comment]
I'd reference the `rxNotify` quark demo from the `rxNotify` component demo. As it stands, there are two places where the promise example markup exists.

**Comment by 100stacks** [comment]
### Updated `rxNotify` Component Demo Page
screen shot 2016-01-14 at 12 49 27 pm

### Updated `rxPromiseNotifications` Quark Demo Page
screen shot 2016-01-14 at 12 09 39 pm

**Comment by 100stacks** [comment]
@CITguy and @glynnis, pulled out the `rxPromiseNotifications` example from the `rxNotify` component demo. :+1: 

Screenshots provided in https://github.com/rackerlabs/encore-ui/pull/1474#issuecomment-171744377 :point_up_2:

File: src/quarks/rxPromiseNotifications/docs/examples/rxPromiseNotifications.simple.html [hunk 3]
```diff
@@ -3,10 +3,6 @@
     <div rx-form flex>
       <rx-form-section>
         <div>
-          <h2>Example</h2>
-          <p>Select <code>Create Service</code> to simulate creating a service, then select
-            <code>Resolve Promise</code> or <code>Reject Promise</code> to simulate success or failure.
-          </p>
           <button class="button" ng-click="addPromise()">Create Promise</button>
           <button class="button positive" ng-click="deferred.resolve();">Resolve Promise</button>
           <button class="button negative" ng-click="deferred.reject();">Reject Promise</button>
```
Comment (CITguy) [inline src/quarks/rxPromiseNotifications/docs/examples/rxPromiseNotifications.simple.html]: The markup for this entire file can be simplified to the following:

```html
<div ng-controller="rxPromiseNotificationsSimpleCtrl">
  <p>
    <button class="button" ng-click="addPromise()">Create Promise</button>
    <button class="button positive" ng-click="deferred.resolve()">Resolve Promise</button>
    <button class="button negative" ng-click="deferred.reject()">Reject Promise</button>
  </p>

  <h3>Demo Stack</h3>
  <rx-notifications stack="demo"></rx-notifications>
</div>
```

Please update.

**Comment by 100stacks** [comment]
@CITguy, `rxPromiseNotifications` example updated. :+1: 

### Demo Page (Main)
screen shot 2016-01-20 at 12 07 38 pm

### Demo Page (Markup Tab)
screen shot 2016-01-20 at 12 08 05 pm

**Comment by CITguy** [comment]
LGTM

**Comment by parlarjb** [comment]
LGTM

**Pull request merged**